server/src/test/java/org/opensearch/search/pipeline/SearchPipelineServiceTests.java

```diff
@@ -235,34 +238,61 @@ public String getTag() {
         public String getDescription() {
             return description;
         }
+
+        @Override
+        public boolean getIgnoreFailure() {
+            return ignoreFailure;
+        }
     }
 
     private static class FakeRequestProcessor extends FakeProcessor implements SearchRequestProcessor {
         private final Consumer<SearchRequest> executor;
 
-        public FakeRequestProcessor(String type, String tag, String description, Consumer<SearchRequest> executor) {
-            super(type, tag, description);
+        public FakeRequestProcessor(String type, String tag, String description, boolean ignoreFailure, Consumer<SearchRequest> executor) {
+            super(type, tag, description, ignoreFailure);
             this.executor = executor;
         }
 
         @Override
         public SearchRequest processRequest(SearchRequest request) throws Exception {
-            executor.accept(request);
+            try {
+                executor.accept(request);
+            } catch (Exception e) {
+                if (this.getIgnoreFailure()) {
+                    return request;
+                } else {
+                    throw e;
+                }
```

Comment:
You shouldn't add the logic to a processor for testing, since we'll want to verify the catch+ignore behavior of the `Pipeline` code. This will swallow the exception before `Pipeline` even gets a chance to see it.

Reply:
remove catch